### PR TITLE
[CI] Add criticality of release to release notes

### PR DESCRIPTION
**Pull request by s3krit**
* [x] Enforces presence of C-labels to designate the importance of a release
* [x] Iterates over them on a release, takes the highest present and includes it in the release notes
* [x] Lists the change(s) that caused the release to be that priority

Devops ticket here: https://github.com/paritytech/devops/issues/569

### CODE REVIEW

**Comment by coriolinus** [comment]
If we mandate a C label, then I'd suggest a `C0-irrelevant` or similar label for PRs which don't affect end-users at all, such as this one, or https://github.com/paritytech/polkadot/pull/1257.

**Comment by s3krit** [comment]
> If we mandate a C label, then I'd suggest a `C0-irrelevant` or similar label for PRs which don't affect end-users at all, such as this one, or #1257.

Think of the label as something that changes the importance of the user updating in a timely manner - the default importance for the user updating is Low. The importance of a release should never be 'irrelevant' (else why make a release at all). We could also have it so that PRs are by default labelled as low importance, but having it as a manual step causes the author of the PR to think a little about the importance of their PR.

**Comment by gavofyork** [comment]
> Think of the label as something that changes the importance of the user updating in a timely manner - the default importance for the user updating is Low. The importance of a release should never be 'irrelevant' (else why make a release at all). We could also have it so that PRs are by default labelled as low importance, but having it as a manual step causes the author of the PR to think a little about the importance of their PR.

agree on all points.

**Comment by s3krit** [comment]
There can be some confusing bash in here due to the use of associativee arrays, but I'm happy to walk any reviewers through it.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -157,9 +157,7 @@ build-wasm-release:
   <<:                             *compiler_info
   # Note: We likely only want to do this for tagged releases, hence the 'only:'
   only:
-    # FIXME remove when ready to merge
```
Comment (gabreal) [inline .gitlab-ci.yml:160]: don't forget

**Pull request merged**